src/molecules/zaboImage/ZaboImage.tsx

```diff
@@ -0,0 +1,105 @@
+import { useState } from "react";
+import Skeleton from "react-loading-skeleton";
+import Image from "src/atoms/image/Image";
+import colorSet from "src/styles/colorSet";
+import styled, { css } from "styled-components";
+
+const HoverImage = styled(Image)<{ isHover: boolean; shadowColor: string }>`
+  transition: 0.1s;
+  z-index: 1;
+  ${({ isHover, shadowColor }) => {
+    if (isHover) {
+      return css`
+        transform: translateY(-10px);
+        box-shadow: ${colorSet.primary}40 0px 5px, ${shadowColor}30 0px 10px,
+          ${shadowColor}20 0px 15px, ${shadowColor}10 0px 20px,
+          ${shadowColor}05 0px 25px;
+      `;
+    }
+  }}
+`;
+
+const HeightOriginIMG = styled(HoverImage)<{
+  size: number;
+}>`
+  height: ${({ size }) => `${size}px`};
+  min-width: ${({ size }) => `calc(${size}px / 1.5`};
+  max-width: ${({ size }) => `calc(2 * ${size}px)`};
+  object-position: "center";
+`;
+
+const WidthOriginIMG = styled(HoverImage)<{
+  size: number;
+}>`
+  width: ${({ size }) => `${size}px`};
+  min-height: ${({ size }) => `${size}px`};
+  max-height: ${({ size }) => `calc(2 * ${size}px)`};
+`;
+
+interface SkeletonRendererProps {
+  width: number;
+  height: number;
+}
+
+const SkeletonRenderer = ({ width, height }: SkeletonRendererProps) => {
+  return (
+    <Skeleton
+      style={{
+        borderRadius: "5px",
+        width: width ? `${width}px` : undefined,
+        height: height ? `${height}px` : undefined,
+      }}
+    />
+  );
+};
+
+type ZaboImageOrigin = "height" | "width";
+
+export interface ZaboImageProps {
+  src: string;
+  origin: ZaboImageOrigin;
+  size: number;
+  isHover: boolean;
+}
+
+const ZaboImage = ({ src, origin, size, isHover }: ZaboImageProps) => {
+  const [isLoading, setIsLoading] = useState(true);
+  const handleImageLoad = () => {
+    setIsLoading(false);
+  };
+
+  switch (origin) {
+    case "height":
+      return (
+        <>
+          {isLoading && <SkeletonRenderer width={size} height={size} />}
+          <HeightOriginIMG
+            src={src}
+            size={size}
+            isHover={isHover}
+            shadowColor={colorSet.primary}
+            isLoading={isLoading}
+            onLoad={handleImageLoad}
+            borderRadius="5px"
+          />
```

Comment:
```suggestion
          {isLoading ? <SkeletonRenderer width={size} height={size} />
          :
          <HeightOriginIMG
            src={src}
            size={size}
            isHover={isHover}
            shadowColor={colorSet.primary}
            isLoading={isLoading}
            onLoad={handleImageLoad}
            borderRadius="5px"
          />}
```

이러는 게 더 깔끔하고 성능상 나아보여요

Reply:
@dohyun-ko @2paperstar 
스켈레톤이 왜 안 나오나 했더니, 저 상태로 두면 img가 아예 로딩되지 않아서 onLoad가 영원히 호출되지 않는 일이 발생합니다. 
제가 저번에 했을 때는 로딩 안 되었을 때 display none으로 뒀었는데 이 방법이 통했던 이유가 그거였네요

ImageRenderer 컴포넌트를 Image와 ZaboImage 컴포넌트 두 개로 쪼갤 때 저렇게 리팩토링했었는데 display none 방식으로 수정하겠습니다
근데 display none 마음에 안드는데 뭔가 더 좋은 방법이 있으면 알려주세요

Reply:
저 부분만 따로 컴포넌트를 분리하고, 이미지를 programmatically 한 방법으로 불러오는 것도 좋을 것 같습니다.